README.md

```diff
@@ -39,18 +42,97 @@ let waiter = buildWaiter('friend-waiter');
 
 export default class Friendz extends Component {
   didInsertElement() {
-    waiter.beginAsync(this);
+    let token = waiter.beginAsync();
+
+    someAsyncWork()
+      .then(() => {
+        //... some work
+      })
+      .finally(() => {
+        waiter.endAsync(token);
+      });
+  }
+}
+```
+
+#### Custom Tokens
+
+The `beginAsync` method also can receive a custom token, if you need to specifically define a token to represent your async operation. _Note_ - due to the nature of
+async operations, it's impossible to guarantee which order the `endAsync` calls will run. If you are making multiple `beginAsync`/`endAsync` calls wiithin a single
```

Comment:
What I mean to say here is it's not possible to ensure that these calls will not interleave, meaning that two separate pairs of `beginAsync`/`endAsync` won't interleave.

Reply:
Right, but the caller of `beginAsync` and `endAsync` **absolutely** can ensure they won't interleave. It may be hard, but it is absolutely possible...

Reply:
I think I'm focusing on the common case here, where the average caller may not be thinking about the full breadth of their async operations (and yes, they should be!). I don't disagree with you, I just believe that the token system makes them fall into a pit of success in a much easier way.

Reply:
Updated the prose.